server/src/main/java/com/cloud/template/TemplateManagerImpl.java

```diff
@@ -1747,8 +1747,8 @@ public VMTemplateVO createPrivateTemplateRecord(CreateTemplateCmd cmd, Account t
         _accountMgr.checkAccess(caller, null, true, templateOwner);
 
         String name = cmd.getTemplateName();
-        if ((name == null) || (name.length() > 32)) {
-            throw new InvalidParameterValueException("Template name cannot be null and should be less than 32 characters");
+        if ((name == null) || (name.length() > 255)) {
```

Comment:
would be good to define a const ?

Reply:
yes, but that constant would have to be used in more than just this method (also for registiring/editting of a template. That would be a bit beyond the scope of this change.

Reply:
did it anyway, I didn´t implement it in the there APIs though.